Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -647,8 +664,10 @@ google_firestore_v1_Write Serializer::EncodeMutation(
 
 Mutation Serializer::DecodeMutation(
     nanopb::Reader* reader, const google_firestore_v1_Write& mutation) const {
-  Precondition precondition =
-      DecodePrecondition(reader, mutation.current_document);
+  auto precondition = Precondition::None();
+  if (mutation.has_current_document) {
```

Comment:
Note: `DecodePrecondition` actually guards against the case where `current_document` is not set, but I think having an explicit check here is better -- it's more explicit, and also the way `DecodePrecondition` does the check is a little hacky (it checks whether the fields are zeroed).

Reply:
SGTM. This seems like the best we can do given the way nanopb separates the has-marker from the actual value.